controllers/async_controller.go

```diff
@@ -176,7 +176,7 @@ func (r *AsyncReconciler) Reconcile(req ctrl.Request, obj runtime.Object) (resul
 	// Implementations of Ensure() tend to set their outcomes in obj.Status
 	err = r.Status().Update(ctx, obj)
 	if err != nil {
-		r.Telemetry.LogInfoByInstance("status", "failed updating status", req.String())
+		r.Telemetry.LogInfoByInstance("status", "failed updating status : "+err.Error(), req.String())
```

Comment:
This seems to be the fundamental part that is out of the PR's scope. So, please revert it when the pull request is ready for review.

Reply:
Reverted this change.